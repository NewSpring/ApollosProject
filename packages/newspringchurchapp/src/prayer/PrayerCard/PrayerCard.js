import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import moment from 'moment';

import ActionSheet from 'react-native-actionsheet';
import {
  Avatar,
  BodyText,
  Card,
  CardContent,
  H3,
  H5,
  H6,
  PaddedView,
  styled,
  Touchable,
  ButtonLink,
  ChannelLabel,
} from '@apollosproject/ui-kit';

const HeaderView = styled(({ theme }) => ({
  display: 'flex',
  justifyContent: 'space-between',
  flexDirection: 'row',
  alignItems: 'flex-end',
  marginHorizontal: theme.sizing.baseUnit,
  marginTop: theme.sizing.baseUnit * 0.5,
}))(View);

const GreyH3 = styled(({ theme }) => ({
  color: theme.colors.lightTertiary,
}))(H3);

const GreyH5 = styled(({ theme }) => ({
  color: theme.colors.text.tertiary,
}))(H5);

const GreyH6 = styled(({ theme }) => ({
  color: theme.colors.text.tertiary,
}))(H6);

const StyledCardContent = styled(() => ({
  alignItems: 'center',
}))(CardContent);

const UserHeader = styled(({ theme }) => ({
  alignItems: 'center',
  marginBottom: theme.sizing.baseUnit * 1.5,
}))(View);

const StyledBodyText = styled(() => ({
  textAlign: 'center',
}))(BodyText);

class PrayerCard extends PureComponent {
  handleShowActionSheet = () => {
    this.ActionSheet.show();
  };

  render() {
    const {
      interactive,
      showHelp,
      header,
      avatarSource,
      avatarSize,
      created,
      name,
      source,
      prayer,
      options,
    } = this.props;

    // add a cancel button
    const buttons = options || [];
    buttons.push({ title: 'Cancel', method: null, destructive: false });

    const handleOnPress = (index) => {
      const buttonMethods = buttons.map((option) => option.method);
      // don't use this if they clicked "cancel"
      if (index === buttonMethods.length - 1) return;
      buttonMethods[index]();
    };

    return (
      <Card>
        {interactive ? (
          <HeaderView>
            <GreyH5>{created ? moment(created).fromNow() : ''}</GreyH5>
            {buttons ? (
              <ButtonLink onPress={this.handleShowActionSheet}>
                <GreyH3>...</GreyH3>
              </ButtonLink>
            ) : null}
            <ActionSheet
              ref={(sheet) => {
                this.ActionSheet = sheet;
              }}
              options={buttons.map((option) => option.title)}
              cancelButtonIndex={buttons.length} // ActionSheet only allows for one destructive button // this will only make the first option listed as destructive turn red
              destructiveButtonIndex={buttons
                .map((option) => option.destructive)
                .indexOf(true)}
              onPress={(index) => handleOnPress(index)}
            />
          </HeaderView>
        ) : null}
        <StyledCardContent>
          {header ? (
            <UserHeader>
              <Avatar source={avatarSource} size={avatarSize} />
              <H3>Pray For {name}</H3>
              {source ? <GreyH6>{source}</GreyH6> : null}
            </UserHeader>
          ) : null}
          <StyledBodyText>{prayer}</StyledBodyText>
          {showHelp ? (
            <PaddedView>
              <Touchable onPress={() => {}}>
                <ChannelLabel icon="information" label="How to Pray?" />
              </Touchable>
            </PaddedView>
          ) : null}
        </StyledCardContent>
      </Card>
    );
  }
}

PrayerCard.propTypes = {
  interactive: PropTypes.bool,
  showHelp: PropTypes.bool,
  header: PropTypes.bool,
  avatarSource: PropTypes.shape({ uri: PropTypes.string }),
  avatarSize: PropTypes.string,
  created: PropTypes.string,
  name: PropTypes.string,
  prayer: PropTypes.string,
  source: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      method: PropTypes.func,
      destructive: PropTypes.bool,
    })
  ),
};

PrayerCard.defaultProps = {
  interactive: true,
  showHelp: true,
  header: true,
  avatarSize: 'small',
  name: 'Request',
  source: 'Web',
};

export default PrayerCard;
