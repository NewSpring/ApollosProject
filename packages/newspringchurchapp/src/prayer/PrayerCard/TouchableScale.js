import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Animated, TouchableWithoutFeedback, Vibration } from 'react-native';

class TouchableScale extends Component {
  static propTypes = {
    minScale: PropTypes.number,
    springConfig: PropTypes.shape({}),
    active: PropTypes.bool,
    style: PropTypes.any, // eslint-disable-line
    children: PropTypes.any, // eslint-disable-line
  };

  static defaultProps = {
    minScale: 0.95,
    springConfig: {
      speed: 20,
    },
  };

  scale = new Animated.Value(this.props.active ? this.props.minScale : 1);

  border = new Animated.Value(this.props.active ? 1 : 0);

  opacity = new Animated.Value(this.props.active ? 0 : 1);

  animatedStyle = {
    transform: [{ scale: this.scale }],
    borderTopWidth: this.border.interpolate({
      inputRange: [0, 0.25, 0.5, 0.75, 1],
      outputRange: [0, 1, 1, 1, 1],
    }),
    borderRightWidth: this.border.interpolate({
      inputRange: [0, 0.25, 0.5, 0.75, 1],
      outputRange: [0, 0, 1, 1, 1],
    }),
    borderBottomWidth: this.border.interpolate({
      inputRange: [0, 0.25, 0.5, 0.75, 1],
      outputRange: [0, 0, 0, 1, 1],
    }),
    borderLeftWidth: this.border.interpolate({
      inputRange: [0, 0.25, 0.5, 0.75, 1],
      outputRange: [0, 0, 0, 0, 1],
    }),
    opacity: this.opacity,
  };

  handlePressIn = () => {
    Vibration.vibrate([0, 2000, 2000]);
    Animated.sequence([
      Animated.parallel([
        Animated.spring(this.scale, {
          toValue: this.props.minScale,
          isInteraction: false,
          ...this.props.springConfig,
        }),

        Animated.timing(this.border, {
          toValue: 1,
          duration: 3000,
        }),
      ]),
      Animated.timing(this.opacity, {
        toValue: 0,
        duration: 1000,
      }),
    ]).start();
  };

  handlePressOut = () => {
    Vibration.cancel();
    Animated.sequence([
      Animated.timing(this.opacity, {
        toValue: 1,
        duration: 1000,
      }),
      Animated.parallel([
        Animated.spring(this.scale, {
          toValue: 1,
          isInteraction: false,
          ...this.props.springConfig,
        }),
        Animated.timing(this.border, {
          toValue: 0,
          duration: 3000,
        }),
      ]),
    ]).start();
  };

  render() {
    const { minScale, style, children, ...touchableProps } = this.props;
    return (
      <TouchableWithoutFeedback
        {...touchableProps}
        onPressIn={this.handlePressIn}
        onPressOut={this.handlePressOut}
      >
        <Animated.View style={[this.animatedStyle, style]}>
          {children}
        </Animated.View>
      </TouchableWithoutFeedback>
    );
  }
}

export default TouchableScale;
