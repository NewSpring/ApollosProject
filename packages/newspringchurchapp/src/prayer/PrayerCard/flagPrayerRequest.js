import gql from 'graphql-tag';

export default gql`
  mutation flagPrayerRequest($parsedId: String!) {
    flagRequest(id: $parsedId) {
      id
    }
  }
`;
