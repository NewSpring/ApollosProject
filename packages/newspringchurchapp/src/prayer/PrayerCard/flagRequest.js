import gql from 'graphql-tag';

export default gql`
  mutation flagRequest($parsedId: String!) {
    flagRequest(id: $parsedId) {
      id
    }
  }
`;
