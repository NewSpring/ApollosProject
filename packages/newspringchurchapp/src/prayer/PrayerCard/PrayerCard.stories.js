import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { PaddedView } from '@apollosproject/ui-kit';

import PrayerCard from './PrayerCard';

const cardData = {
  id: 'PrayerRequest:519df72a48836aa524655df9cceb6989',
  imageSource: {
    uri: 'https://fillmurray.com/400/600',
  },
  name: 'Bill',
  text:
    'I’m alright. Nobody worry ’bout me. Why you got to gimme a fight? Can’t you just let it be?',
  source: 'Anderson',
  prayerHelpContent: {
    id: 'MediaContentItem:ebb48774ff5a859981a32c1e38f7c2ec',
    title: 'Learning how to pray like Jesus',
  },
};

storiesOf('PrayerCard', module).add('Example', () => (
  <PaddedView>
    <PrayerCard
      actionsEnabled={false}
      imageSource={cardData.imageSource}
      name={cardData.name}
      text={cardData.text}
      source={cardData.source}
      prayerHelpContent={cardData.prayerHelpContent}
    />
  </PaddedView>
));

storiesOf('PrayerCard', module).add('actionsEnabled', () => (
  <PaddedView>
    <PrayerCard
      actionsEnabled
      id={cardData.id}
      imageSource={cardData.imageSource}
      name={cardData.name}
      text={cardData.text}
      source={cardData.source}
    />
  </PaddedView>
));
