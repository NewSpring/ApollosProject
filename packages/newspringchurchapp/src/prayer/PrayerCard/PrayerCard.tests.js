import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';

import PrayerCard from './PrayerCard';

const prayer = {
  person: {
    photo: {
      uri: 'https://fillmurray.com/400/600',
    },
  },
  firstName: 'Bill',
  text:
    'I want to go back to school for my PTA degree next year, but I’m afraid of not getting accepted. I have a great job right now but I’m not very happy. Pray my plans for school align with God’s plans for me.',
  campus: { name: 'Anderson' },
};

describe('the PrayerCard component', () => {
  it('renders a prayer card', () => {
    const tree = renderer.create(
      <Providers>
        <PrayerCard prayer={prayer} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
