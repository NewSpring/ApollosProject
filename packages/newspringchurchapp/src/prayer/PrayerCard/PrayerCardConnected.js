import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';

import incrementRequestPrayed from '../data/mutations/incrementRequestPrayed';
import PrayerCard from './PrayerCard';

const PrayerCardConnected = memo(({ prayerId, ...props }) => (
  <Mutation mutation={incrementRequestPrayed}>
    {(incrementPrayed) => (
      <PrayerCard
        prayerId={prayerId}
        incrementPrayer={async () => {
          await incrementPrayed({
            variables: {
              parsedId: prayerId,
            },
          });
        }}
        {...props}
      />
    )}
  </Mutation>
));

PrayerCardConnected.propTypes = {
  prayerId: PropTypes.string,
};

PrayerCardConnected.displayName = 'PrayerCardConnected';

export default PrayerCardConnected;
