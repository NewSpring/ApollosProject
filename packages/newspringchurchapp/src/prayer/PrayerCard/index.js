import PrayerCard from './PrayerCard';

export PrayerCardConnected from './PrayerCardConnected';
export default PrayerCard;
