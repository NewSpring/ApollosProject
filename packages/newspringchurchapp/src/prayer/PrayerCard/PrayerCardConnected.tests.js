import React from 'react';
import { renderWithApolloData } from 'newspringchurchapp/src/utils/testUtils';

import Providers from 'newspringchurchapp/src/Providers';
import flagPrayerRequest from './flagPrayerRequest';
import PrayerCardConnected from './PrayerCardConnected';

const prayerId = [
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4789',
  },
];
const cardData = {
  imageSource: {
    uri: 'https://fillmurray.com/400/600',
  },
  name: 'Bill',
  text:
    'I want to go back to school for my PTA degree next year, but I’m afraid of not getting accepted. I have a great job right now but I’m not very happy. Pray my plans for school align with God’s plans for me.',
  source: 'Anderson',
};

describe('the PrayerCardConnected component', () => {
  it('renders prayer card connected to data', async () => {
    const mock = {
      request: {
        query: flagPrayerRequest,
      },
      response: {
        data: {
          prayerId,
        },
      },
    };
    const tree = await renderWithApolloData(
      <Providers mocks={[mock]} addTypename={false}>
        <PrayerCardConnected
          imageSource={cardData.imageSource}
          name={cardData.name}
          text={cardData.text}
          source={cardData.source}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
