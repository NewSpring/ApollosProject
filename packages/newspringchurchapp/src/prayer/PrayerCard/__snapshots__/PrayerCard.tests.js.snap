// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the PrayerCard component renders a prayer card 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 16,
          "marginHorizontal": 0,
          "marginVertical": 0,
          "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#6bac43",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#505050",
                  "fontFamily": "Colfax-Bold",
                  "fontSize": 16,
                  "lineHeight": 23.04,
                  "textAlign": "right",
                }
              }
            >
              ...
            </Text>
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderRadius": 20,
                "height": 40,
                "justifyContent": "flex-end",
                "marginBottom": 10,
                "marginRight": 10,
                "overflow": "hidden",
                "width": 40,
              }
            }
            themeSize={40}
          >
            <View
              animate={null}
              customAnimate={null}
              onReady={false}
              style={
                Object {
                  "aspectRatio": 1,
                  "backgroundColor": "#dddddd",
                  "width": "100%",
                }
              }
            >
              <Image
                onLoad={[Function]}
                source={
                  Array [
                    Object {
                      "cache": "force-cache",
                      "uri": "https://fillmurray.com/400/600",
                    },
                  ]
                }
                style={
                  Object {
                    "backgroundColor": "#dddddd",
                    "borderRadius": 20,
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
                themeSize={40}
              />
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
              }
            }
          >
             Pray For 
            Bill
          </Text>
          <Text
            style={
              Object {
                "color": "#505050",
                "fontFamily": "Colfax-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
              }
            }
          >
            Anderson
          </Text>
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Regular",
                "fontSize": 16,
                "lineHeight": 26,
                "marginTop": 24,
              }
            }
          >
            I want to go back to school for my PTA degree next year, but I’m afraid of not getting accepted. I have a great job right now but I’m not very happy. Pray my plans for school align with God’s plans for me.
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              flexed={false}
              style={
                Object {
                  "alignItems": "center",
                  "flex": null,
                  "flexDirection": "row",
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={19.2}
                bbWidth={19.2}
                height={19.2}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    undefined,
                    null,
                    Object {
                      "flex": 0,
                      "height": 19,
                      "width": 19,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={19.2}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                      "fillOpacity",
                      "fillRule",
                      "stroke",
                      "strokeWidth",
                      "strokeOpacity",
                      "strokeDasharray",
                      "strokeDashoffset",
                      "strokeLinecap",
                      "strokeLinejoin",
                      "strokeMiterlimit",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  x={0}
                  y={0}
                >
                  <RNSVGPath
                    d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm2.2 15.24c-.54.2-2.44 1.13-3.54.16-.33-.3-.5-.66-.5-1.1 0-.83.28-1.56.77-3.3.1-.33.2-.76.2-1.1 0-.58-.23-.73-.83-.73-.3 0-.62.1-.9.2l.15-.65c.65-.27 1.48-.6 2.18-.6 1.06 0 1.84.53 1.84 1.53 0 .3-.05.8-.16 1.15l-.6 2.15c-.1.44-.3 1.4 0 1.68s1.2.13 1.6-.06l-.14.67zm-1-8.57c-.68 0-1.24-.56-1.24-1.25 0-.7.56-1.25 1.25-1.25.7 0 1.3.56 1.3 1.25 0 .7-.55 1.25-1.24 1.25z"
                    fill={
                      Array [
                        0,
                        4286940549,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    x={0}
                    y={0}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <View
                style={
                  Object {
                    "paddingHorizontal": 4,
                  }
                }
                withIcon="information"
              >
                <Text
                  style={
                    Object {
                      "color": "#858585",
                      "fontFamily": "Colfax-Bold",
                      "fontSize": 12,
                      "lineHeight": 17.28,
                    }
                  }
                  tint="#858585"
                >
                  How to Pray?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "paddingVertical": 16,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                bordered={false}
                disabled={false}
                pill={true}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#6bac43",
                    "borderColor": "#6bac43",
                    "borderRadius": 48,
                    "borderWidth": 2,
                    "elevation": 2,
                    "flexDirection": "row",
                    "height": 48,
                    "justifyContent": "center",
                    "opacity": 1,
                    "overflow": "hidden",
                    "paddingHorizontal": 16,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Colfax-Bold",
                      "fontSize": 16,
                      "lineHeight": 23.04,
                    }
                  }
                >
                  Start Praying for Bill
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
